app.ts

```diff
@@ -0,0 +1,238 @@
+				clearTimeout(Load);
+				Load = setTimeout(async () => {
+					const recordCount = await getRecordCountCall();
+					if (nextToId >= recordCount - 1) {
+						const fromId = recordCount - 1 - (calculateToId(previous[0]) - previous[0]);
+						const toId = recordCount - 1;
+						previous = await LoadRecordsData(fromId, toId);
+					} else {
+						previous = await LoadRecordsData(previous[0], nextToId)
+					}
+				}, 10);
+			} catch (error) {
+				// throw new Error("Error" + error);
+			}
+		}
+		// Handlers
+		async function LoadPageContent(fromID: number, toID: number): Promise<number[]> {
+
+			try {
+				let DisplayContent = "";
+				const columns = await getColumnNamesCall();
+				for (const column of columns) {
+					DisplayContent += `<th align="center">${column}</th>`;
+					$("#wrapper-table-header-row").empty();
+					$("#wrapper-table-header-row").append(DisplayContent);
+				}
+			} catch (error) {
+				console.log("Error <No Data>" + error);
+			}
+			return await LoadRecordsData(fromID, toID);
+		}
+		function ConvertNumber(input: string | number, parseAsInt: boolean = true): number {
+			switch (typeof input) {
+				case ('string'):
+					if (parseAsInt == true) {
+						return parseInt(input as string);
+					}
+					return parseFloat(input as string);
+				case ("number"):
+					return input as number;
+				default:
+					return 0;
+			}
+			throw new Error("Error");
+		}
+		function calculateToId(fromId: number): number {
+			const possibleRecordsData = Math.floor((window.innerHeight - ($("#form-content").innerHeight() as number)) / 37);
+			const possibleId = fromId + possibleRecordsData;
+
+			let recordDisplayset = 0;
+			switch (recordDisplayset) {
+				case 0:
+					window.innerHeight <= 646;
+					break;
+				case 1:
+					window.innerHeight <= 969;
+					break;
+				case 2:
+					window.innerHeight <= 1938;
+					break;
+				default:
+					recordDisplayset = 15;
+					break;
+			}
+			return recordDisplayset + possibleId;
+		}
+		function nextPageResize(previous: number[]): number {
+			const fromID = ConvertNumber(previous.sort((a, b) => { return a - b })[0]);
+			const toID = ConvertNumber(previous.sort((a, b) => { return a - b })[1]);
+			const documentHeight = $(window).innerHeight() as number - ($(`#table-row-${fromID}`).height() as number);
+			for (let i = fromID; i <= toID; i++) {
+				const elementHeightOffset = ($(`#table-row-${i}`).offset() as JQueryCoordinates).top;
+				if (elementHeightOffset < documentHeight) continue;
+				return i;
+			}
+			return toID;
+		}
+		function previousPageResize(previous: number[]): number[] {
+			const toId = calculateToId(previous[0] - (nextPageResize(previous) - previous[0]));
+			return [previous[0] - (nextPageResize(previous) - previous[0]), toId];
+		}
+		window.onload = async () => {
+			try {
+				previous = await LoadPageContent(0, calculateToId(0));
+				$("#previous").click(async () => {
+					const CountData = await getRecordCountCall();
```

Comment:
You are allowed to skip this step. For this project the record count will not change after the initial call.